arm-storsimple8000series/2017-06-01/swagger/storsimple.json

```diff
@@ -2069,362 +2501,3442 @@
+        "provider": {
+          "description": "The localized friendly form of the resource provider name - it is expected to also include the publisher/company responsible. It should use Title Casing and begin with 'Microsoft' for 1st party services.",
+          "type": "string"
+        },
+        "resource": {
+          "description": "The localized friendly form of the resource type related to this action/operation - it should match the public documentation for the resource provider. It should use Title Casing - for examples, please refer to the 'name' section.",
+          "type": "string"
+        },
+        "operation": {
+          "description": "The localized friendly name for the operation, as it should be shown to the user. It should be concise (to fit in drop downs) but clear (i.e. self-documenting). It should use Title Casing and include the entity/resource to which it applies.",
+          "type": "string"
+        },
+        "description": {
+          "description": "The localized friendly description for the operation, as it should be shown to the user. It should be thorough, yet concise - it will be used in tool tips and detailed views.",
+          "type": "string"
+        }
+      }
+    },
+    "AvailableProviderOperationList": {
+      "description": "List of available provider operations.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The value.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AvailableProviderOperation"
+          }
+        },
+        "nextLink": {
+          "description": "The NextLink.",
+          "type": "string"
+        }
+      }
+    },
+    "AvailableProviderOperationProperties": {
+      "description": "Represents properties of AvailableProviderOperation",
+      "type": "object",
+      "properties": {}
+    },
+    "Backup": {
+      "description": "The backup.",
+      "required": [
+        "properties"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseModel"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BackupProperties",
+          "description": "The properties of the backup.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "BackupElement": {
+      "description": "The backup element.",
+      "required": [
+        "elementId",
+        "elementName",
+        "elementType",
+        "sizeInBytes",
+        "volumeName",
+        "volumeContainerId"
+      ],
+      "type": "object",
+      "properties": {
+        "elementId": {
+          "description": "The path ID that uniquely identifies the backup element.",
+          "type": "string"
+        },
+        "elementName": {
+          "description": "The name of the backup element.",
+          "type": "string"
+        },
+        "elementType
```

Comment:
Thanks for the reminder on the other PR, though I'm not sure why these are not proxy-resources, I don't see a conclusion for this in the referred PR, if you have it, please let me know. Sorry that it looks like I missed replying to a comment from you in the other PR. 
Here are your questions from the other one: 
@veronicagg We see the Autorest linter pointing for requirement of this "x-ms-azure-resource" only for Proxy-resources for which PUT call exists. For others, they are created out of POST call and returned only in Get calls or PATCH calls. For e.g., in current PR, 'Device' is a proxy resource, but put call doesn't exists for it (as it's registration and creation happens by device directly talking with service via IdManagement, etc). But, a PATCH operation, namely 'Device_Update' returns it.

[Veronica] The fact that the linter detects some of the instances is because we may not have an accurate way to find these and it helps us giving hints manually look for more. The question comes not just due to the validation rules, but from manual review. 

(a) Is it fine if we mark this field true for all of them? If that is the case, we will then go ahead and mark this as true for all the resources.
[Veronica] If you have a "base" that you're using for all of them, you can mark that one as "x-ms-azure-resource"

(b) Also, from documentation I could find only usage of this field as flattening of code, is there any detailed documentation around that and other usage, if any?
[Veronica] Per the documentaion at https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-azure-resource if the entity is a Resource as per ARM guidelines, then to accurately represent it, it should be tagged. Additionally, this property is used to detect violations for these entities and code generators can decide on any special treatment for them.
@ravbhatnagar pulling you in, to confirm.
I believe Gaurav is planning to ask arm specs to refer to https://github.com/Azure/azure-rest-api-specs-pr/blob/master/common-types/resource-management/v1/types.json in this case, would you think these entities should be "allOf"s this https://github.com/Azure/azure-rest-api-specs-pr/blob/master/common-types/resource-management/v1/types.json#L3 ?